snorkel/labeling/model/label_model.py

```diff
@@ -368,10 +369,124 @@ def predict_proba(self, L: sparse.spmatrix) -> np.ndarray:
+        """
+
+        n, k = Y_prob.shape
+        Y_pred = np.zeros(n)
+        diffs = np.abs(Y_prob - Y_prob.max(axis=1).reshape(-1, 1))
+
+        TOL = 1e-5
+        for i in range(n):
+            max_idxs = np.where(diffs[i, :] < TOL)[0]
+            if len(max_idxs) == 1:
+                Y_pred[i] = max_idxs[0] + 1
+            # Deal with "tie votes" according to the specified policy
+            elif break_ties == "random":
+                Y_pred[i] = np.random.choice(max_idxs) + 1
+            elif break_ties == "abstain":
+                Y_pred[i] = 0
+            else:
+                raise ValueError(f"break_ties={break_ties} policy not recognized.")
+        return Y_pred
+
+    def predict(
+        self, L: sparse.spmatrix, break_ties: Optional[str] = "abstain"
+    ) -> np.ndarray:
+        """Return predicted labels, with ties broken according to policy.
+
+        Policies to break ties include:
+        "abstain": return an abstain vote (0)
+        "random": randomly choose among the tied options
+        NOTE: if break_ties="random", repeated runs may have slightly different results due to difference in broken ties
+
+
+        Parameters
+        ----------
+        L
+            An [n,m] matrix with values in {0,1,...,k}
+        break_ties
+            Policy to break ties when converting probabilistic labels to predictions
+
+        Returns
+        -------
+        np.ndarray
+            An [n,k] array of integer labels
+
+        Example
+        -------
+        >>> L = sparse.csr_matrix([[1, 1, 0], [2, 2, 0], [1, 1, 0]])
+        >>> label_model = LabelModel()
+        >>> label_model.train_model(L)
+        >>> label_model.predict(L)
+        np.array([1, 2, 1])
+        """
+        Y_probs = self.predict_proba(L)
+        Y_p = self._break_ties(Y_probs, break_ties).astype(np.int)
+        return Y_p
+
+    def score(
+        self,
+        L: sparse.spmatrix,
+        Y: np.ndarray,
+        metrics: Optional[List[str]] = ["accuracy"],
+    ) -> Dict[str, float]:
+        """Calculate one or more scores from user-specified and/or user-defined metrics.
+
+        Parameters
+        ----------
+        L
+            An [n,m] matrix with values in {0,1,...,k}
+        Y
+            Gold labels associated with datapoints in L
+        metrics
+            A list of metric names, by default ["accuracy"]
+
+        Returns
+        -------
+        Dict[str, float]
+            A dictionary mapping metric names to metric scores
+
+        Example
+        -------
+        >>> L = sparse.csr_matrix([[1, 1, 0], [2, 2, 0], [1, 1, 0]])
+        >>> label_model = LabelModel()
+        >>> label_model.train_model(L)
+        >>> label_model.score(L, Y=np.array([1, 1, 1]))
+        {'accuracy': 0.66667}
+        >>> label_model.score(L, Y=np.array([1, 1, 1], metrics=["f1"]))
+        {'accuracy': 0.66667}
+        """
+        Y_prob = self.predict_proba(L)
+        Y_pred = self.predict(L)
```

Comment:
This will compute predict_prob twice. Can we add a `return_probs` flag to `predict()`?